swap/src/bin/asb.rs

```diff
@@ -212,3 +232,46 @@ async fn init_monero_wallet(
 
     Ok(wallet)
 }
+
+/// Registers a hidden service for each network.
+/// Note: Once ac goes out of scope, the services will be de-registered.
+async fn register_tor_services(
+    networks: Vec<Multiaddr>,
+    uac: UnauthenticatedConnection,
+) -> Result<AuthenticatedConnection> {
+    let mut ac = uac.into_authenticated_connection().await?;
+
+    let hidden_services_details = networks
+        .iter()
+        .flat_map(|network| {
+            network.iter().map(|protocol| match protocol {
+                Protocol::Tcp(port) => Some((
+                    port,
+                    SocketAddr::new(IpAddr::from(Ipv4Addr::new(127, 0, 0, 1)), port),
+                )),
+                _ => {
+                    // We only care for Tcp for now.
+                    None
+                }
+            })
+        })
+        .filter_map(|details| details)
+        .collect::<Vec<_>>();
+
+    // Note: ideally we have deterministic keys but unfortunately there is no Api
+    // for that yet.
```

Comment:
Seems like any 64-bytes can be turned into a secret key: https://docs.rs/torut/0.1.8/torut/onion/struct.TorSecretKeyV3.html#impl-From%3C%5Bu8%3B%2064%5D%3E

Reply:
Would be nice to derive these from the seed as well I guess.

Reply:
Cool, didn't see that from/into is derived.